hackathon_site/hardware/test_api.py

```diff
@@ -51,6 +51,52 @@ def test_hardware_get_success(self):
         self.assertEqual(expected_response, data["results"][0])
 
 
+class HardwareDetailViewTestCase(SetupUserMixin, APITestCase):
+    def setUp(self):
+        super().setUp()
+
+        self.hardware = Hardware.objects.create(
+            name="name",
+            model_number="model",
+            manufacturer="manufacturer",
+            datasheet="/datasheet/location/",
+            quantity_available=4,
+            max_per_team=1,
+            picture="/picture/location",
+        )
+
+        self.hardware_serializer = HardwareSerializer(self.hardware)
+        self.view = reverse(
+            "api:hardware:hardware-detail", kwargs={"pk": self.hardware.id}
+        )
+
+    def test_user_not_logged_in(self):
+        response = self.client.get(self.view)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_hardware_get_success(self):
+        self._login()
+        expected_response = {
+            "id": self.hardware.id,
+            "name": "name",
+            "categories": [],
```

Comment:
We should test that categories appear properly as well. They should just be a list of primary keys.

Could you add a category, link it to the hardware you added (with `self.hardware.categories.add(<category you created>)`, and make sure the categories list here is something like `[self.category.id]`?